images/cppcheck/Dockerfile.erb

```
@@ -4,7 +4,7 @@ FROM sider/devon_rex_python:master
 #
 # See https://github.com/danmar/cppcheck#gnu-make
 
-ARG CPPCHECK_VERSION=2.0
+ARG CPPCHECK_VERSION=2.1
 ARG LIBZ3_VERSION=4.4.1-1~deb10u1
 USER root
 RUN apt-get update && \
```

If we keep using the old version of libz3 (v4.4.1), just `cp` the z3_version_old.h as;

```suggestion
RUN apt-get update && \
...
    cd cppcheck-${CPPCHECK_VERSION} && \
    cp externals/z3_version_old.h externals/z3_version.h && \
    make install -s \
...
```

At https://sourceforge.net/p/cppcheck/news/2020/06/cppcheck-21/ you will find a comment: "If you have an old z3 library and get compilation problems you will need to add a z3_version.h in externals."

Note that I got a segmentation fault with the latest version of libz3. If we want to update the libz3, we need further investigation.

---

Thank you! Your suggestion succeeds the compilation!
I will add comments to describe the notes above.